cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/AbstractStreamService.java

```diff
@@ -97,4 +112,39 @@ protected ScheduledExecutorService executor() {
     executor = Executors.newSingleThreadScheduledExecutor(Threads.createDaemonThreadFactory("heartbeats-scheduler"));
     return executor;
   }
+
+  /**
+   * Get the size of events ingested by a stream since its creation, in bytes.
+   * @param streamId id of the stream
+   * @return Size of events ingested by a stream since its creation
+   * @throws IOException when getting an error retrieving the metric
+   */
+  protected long getStreamEventsSize(Id.Stream streamId) throws IOException {
+    MetricDataQuery metricDataQuery = new MetricDataQuery(
+      0L, TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis()),
+      Integer.MAX_VALUE, "system.collect.bytes",
+      MetricType.COUNTER,
+      ImmutableMap.of(Constants.Metrics.Tag.NAMESPACE, streamId.getNamespaceId(),
+                      Constants.Metrics.Tag.STREAM, streamId.getName()),
+      ImmutableList.<String>of()
+    );
+
+    try {
+      Collection<MetricTimeSeries> metrics = metricStore.query(metricDataQuery);
+      if (metrics == null || metrics.isEmpty()) {
+        // Data is not yet available, which means no data has been ingested by the stream yet
+        return 0L;
+      }
+
+      MetricTimeSeries metric = metrics.iterator().next();
+      List<TimeValue> timeValues = metric.getTimeValues();
+      if (timeValues == null || timeValues.size() != 1) {
+        throw new IOException("Should collect exactly one time value");
+      }
+      return timeValues.get(0).getValue();
+    } catch (Exception e) {
+      Throwables.propagateIfInstanceOf(e, IOException.class);
+      throw new IOException(e);
```

Comment:
what is the behavior when this happens? Should make sure we don't die or anything like that